packages/app/src/components/authorize/ApprovalStateBadge.tsx

```diff
@@ -0,0 +1,13 @@
+import { FC } from "react";
+import { Spinner } from "react-bootstrap";
+import useStore from "../../store";
+
+const ApprovalStateBadge: FC = () => {
+  const approvalState = useStore((state) => state.approvalState);
+
+  return (
+    <>{approvalState !== "None" && approvalState !== "Success" && <Spinner animation="border" variant="primary" />}</>
+  );
+};
+
+export default ApprovalStateBadge;
```

Comment:
Can be removed entirely state needs to be on the button

Reply:
I thought we agreed on doing it the uniswap style, but we can do it on the button if that's what you prefer

Reply:
But I think it would nice anyway to have something in the navbar to inform the user that something is in progress, even when not seeing the button which started the action

Reply:
Sure but that's definitely not the right way to do it, that's what the transaction banner and modal are for. 

Alternatively: https://usedapp.readthedocs.io/en/latest/guide.html#notifications